Adding examples for remove connections for 2 circuits

KeremKurban opened the pull request:
1 circuit for sscx (multipopulation) and a test circuit with several nodes and edges (multipop as well)

KeremKurban commented:
We can squash commits to have linear commit history

chr-pok commented:
Thanks for the PR! Here a couple of suggestions, since I find the examples rather complicated to follow:

- Please specify clearly where a user should start with an example, as there are Jupyter notebooks and multiple readme files. Better use either a single notebook or a single .md file (whatever you prefer) to guide a user through an example.
- Each example should have a clear structure and only contain the minimum number of files absolutely required. So better don't include circuit configs, node sets, output folders, etc. Example outputs and structural comparison plots could/should be integrated directly into the notebook/.md file for explaining the result.
- I think it would be better to use only one circuit for each example, so either SSCx or the small test circuit (because the users can rerun the example anyway with whatever circuit if they like).
- If you prefer to use and include the test circuit directly in the repo, we should
  - make sure that we have permission to publish that circuit (or we should better link the original source, if available)
  - better put the circuit into a separate folder at the level of `/examples` (instead of within a specific example subfolder), so the same test circuit can be easily re-used in multiple examples
- I noticed there is still `output_path` and `working_dir` in some of the manipulation configs (the ones under `test_circuit`) but which is only needed in model configs.
- **Important:** We need to include copyright information in all source/text files
  - For .py files and Jupyter notebooks: Put a statement at the beginning of each file, as in e.g. `NullManipulationExample.ipynb`
  - For readme files: `Copyright (c) 2024 Blue Brain Project/EPFL` at the end of each file

KeremKurban commented:
@chr-pok i added a README under conn_removal/ and specific instructions under each notebook. Moved the example circuit as you said. removed unnecessary files, removed working_dir unless required. i think it was required in str comparison based on the demo. finally added copyright in markdown and notebook files

chr-pok commented:
Thanks @KeremKurban for restructuring and simplifying the example, looks good now! Just one question, do we really need to include all log and output folders in the repo?

KeremKurban commented:
We can remove them as its tested.

chr-pok commented:
Ok, let's keep the struct comparison results as examples, this could be informative for the user as what to expect. But better remove all log files/folders, as there are quite a lot of them.

chr-pok commented:
And I just noticed, we need to add proper reference whenever we mentioning the test circuit from BlueCelluLab, i.e., citation and link to that repo, following Apache 2 license requirements.

KeremKurban commented:
Since BlueCelluLab doesnt have citation info yet, i added the repository link under example circuits

chr-pok commented:
Ah, right, there is no citation indeed, I think putting the link is perfectly fine then, thanks! Last thing is still to remove the .log files, as there are still hundreds of them in output/logs subfolders.

KeremKurban commented:
removed

chr-pok commented on connectome_manipulator/connectome_manipulation/manipulation/conn_removal.py:
@KeremKurban: I think this fix to `conn_removal` is no longer required here and can be reverted. I found and fixed the problem with empty data splits at another location (see my latest commit), as it may affect other operations as well. The problem was that SNAP returns an empty dataframe with 'object' dtypes for empty selections (instead of the actual dtypes) which is causing problems down the line. Please verify that my fix is now working for you as well! Thanks!

KeremKurban commented:
@chr-pok i tested the new code with the small circuit and adapted the changes into my branch.
